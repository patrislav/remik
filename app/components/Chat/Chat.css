.chat {
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
}

.listWrapper {
  overflow-y: auto;
}

.composerWrapper {

}
