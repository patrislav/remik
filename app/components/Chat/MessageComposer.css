.composer {
  width: 100%;
  height: 100%;
  display: flex;
}

.input {
  flex: 1;
}

.submit {
  flex: none;
}
