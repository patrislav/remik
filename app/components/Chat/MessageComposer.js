import React, {PropTypes} from 'react'
import Button from '../Button'
import TextArea from '../TextArea'

const ENTER_KEY_CODE = 13

class MessageComposer extends React.Component {
  static propTypes = {
    onCompose: PropTypes.func
  }

  static defaultProps = {
    onCompose: () => {}
  }

  state = {
    value: ''
  }

  render() {
    const { value } = this.state
    return (
      <div>
        <TextArea
          value={value}
          onChange={::this.handleChange}
          onKeyDown={::this.handleKeyDown}
        />
        <Button onClick={::this.handleCompose}>Send</Button>
      </div>
    )
  }

  handleChange(value) {
    this.setState({ value })
  }

  handleCompose() {
    const { onCompose } = this.props
    const { value } = this.state
    this.setState({ value: '' })
    onCompose(value)
  }

  handleKeyDown(event) {
    if (event.keyCode === ENTER_KEY_CODE) {
      event.preventDefault()
      this.handleCompose()
    }
  }
}

export default MessageComposer
