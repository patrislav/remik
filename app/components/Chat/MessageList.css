.wrapper {
  width: 100%;
  height: 100%;
}

.list {
  list-style-type: none;
  margin: 0;
  padding: 0.3rem;
}

.item {

}
