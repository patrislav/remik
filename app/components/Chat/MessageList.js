import React, {PropTypes} from 'react'

const propTypes = {
  messages: PropTypes.array
}

const defaultProps = {
  messages: []
}

const MessageList = ({ messages }) => (
  <ul>
    {messages.map((message, i) =>
      <li key={i}>{message}</li>
    )}
  </ul>
)

MessageList.propTypes = propTypes
MessageList.defaultProps = defaultProps

export default MessageList
