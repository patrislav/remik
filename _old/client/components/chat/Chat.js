import React, {Component, PropTypes} from 'react'
import {connect} from 'react-redux'
import Immutable from 'immutable'
import {getTypingUsers} from '../../helpers'

import MessageComposer from './MessageComposer'
import MessageView from './MessageView'
import TypingIndicator from './TypingIndicator'

@connect(state => ({
  messages: state.chat.get('messages'),
  typingUsers: getTypingUsers(state),
  typingUpdatedAt: state.chat.get('typingUpdatedAt')
}))
export default class Chat extends Component {
  static propTypes = {
    messages: PropTypes.instanceOf(Immutable.List).isRequired,
    typingUsers: PropTypes.instanceOf(Immutable.List).isRequired,
    onTyping: PropTypes.func,
    onMessage: PropTypes.func,
    dispatch: PropTypes.func.isRequired
  }

  constructor(props) {
    super(props)
    this.state = {
      typingUsers: this.props.typingUsers
    }
  }

  componentDidUpdate() {
    this._scrollToBottom()
  }

  componentDidMount() {
    this.interval = setInterval(() => {
      this.props.dispatch({ type: 'chat.typingUpdate' })
    }, 400)
  }

  componentWillUnmount() {
    clearInterval(this.interval)
  }

  render() {
    return (
      <section className="chat">
        <div className="content">
          <ul className="messageList" ref="messageList">
            {this.props.messages.map((message, i) =>
              <MessageView
                key={i}
                user={message.user}
                message={message.message}
                code={message.code}
                />
            )}
            {<TypingIndicator users={this.props.typingUsers} />}
          </ul>
          <MessageComposer
            onChange={this.props.onTyping}
            onSubmit={this.props.onMessage}
            />
        </div>
      </section>
    )
  }

  _scrollToBottom = () => {
    let ul = this.refs.messageList
    ul.scrollTop = ul.scrollHeight
  }
}
