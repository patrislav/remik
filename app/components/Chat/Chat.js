import React, {PropTypes} from 'react'
import {connect} from '../../lib/rxstate'
import MessageComposer from './MessageComposer'
import MessageList from './MessageList'
import {compose$} from '../../modules/chat'

const propTypes = {
  onCompose: PropTypes.func,
  messages: PropTypes.array
}

const defaultProps = {
  onCompose: () => {},
  messages: []
}

export const Chat = ({ onCompose, messages }) => (
  <section>
    <MessageList messages={messages} />
    <MessageComposer onCompose={onCompose} />
  </section>
)

Chat.propTypes = propTypes
Chat.defaultProps = defaultProps

export default connect(state => ({
  messages: state.chat.messages,
  onCompose: message => compose$.next(message),
}))(Chat)
