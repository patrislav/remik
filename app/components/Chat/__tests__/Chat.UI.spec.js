import React from 'react'
import {shallow} from 'enzyme'
import {shallowToJson} from 'enzyme-to-json'

import {Chat} from '../Chat'

describe('<Chat />', () => {
  it('renders correct UI', () => {
    const wrapper = shallow(<Chat />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
