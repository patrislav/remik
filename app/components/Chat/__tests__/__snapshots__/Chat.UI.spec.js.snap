exports[`<Chat /> renders correct UI 1`] = `
<section>
  <MessageList
    messages={Array []} />
  <MessageComposer
    onCompose={[Function]} />
</section>
`;
